src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +30,221 @@ protected String getRequestMethod() {
     @Override
     @Test
     public void testExecute() throws Exception {
-        AccountAttributes student1 = typicalBundle.accounts.get("student1InCourse1");
-
-        StudentProfileAttributes student1InCourse1Profile = typicalBundle.profiles.get("student1InCourse1");
-        testActionSuccess(student1, student1InCourse1Profile, "Typical case");
-
-        testActionInMasquerade(student1);
-
-        student1 = typicalBundle.accounts.get("student1InTestingSanitizationCourse");
-        StudentProfileAttributes student1InTestingSanitizationCourseProfile =
-                typicalBundle.profiles.get("student1InTestingSanitizationCourse");
-        // simulate sanitization that occurs before persistence
-        student1.sanitizeForSaving();
-        testActionSuccess(student1, student1InTestingSanitizationCourseProfile,
-                "Typical case: attempted script injection");
+        // See test cases below.
     }
 
-    private void testActionSuccess(AccountAttributes student, StudentProfileAttributes studentProfileAttributes,
-                                   String caseDescription) {
-        loginAsStudent(student.googleId);
-
-        ______TS(caseDescription);
-
+    @Test
+    public void testExecute_withExistingProfile_shouldReturnCorrectProfile() {
+        AccountAttributes student1InCourse1 = typicalBundle.accounts.get("student1InCourse1");
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = student1InCourse1.name;
+        loginAsStudent(student1InCourse1.googleId);
         String[] submissionParams = new String[] {
-                Const.ParamsNames.STUDENT_ID, student.googleId,
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+                Const.ParamsNames.STUDENT_ID, student1InCourse1.googleId,
         };
+
         GetStudentProfileAction action = getAction(submissionParams);
         JsonResult result = getJsonResult(action);
 
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
-        StudentProfile output = (StudentProfile) result.getOutput();
-        output.getStudentProfile().modifiedDate = null; // ignore this field for testing
-        assertEquals(studentProfileAttributes.toString(), output.getStudentProfile().toString());
-        assertEquals(student.getName(), output.getName());
+        StudentProfileData actualProfile = (StudentProfileData) result.getOutput();
+        assertEquals(expectedName, actualProfile.getName());
```

Comment:
Perhaps abstract these assertions out to a method? 
testEquals(StudentProfileData expected, StudentProfileData actual) {
...
}